pallets/regulated-assets/src/lib.rs

```diff
@@ -0,0 +1,283 @@
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+
+    use super::*;
+    use common::{Balance, DEFAULT_BALANCE_PRECISION};
+    use frame_support::pallet_prelude::{OptionQuery, ValueQuery, *};
+    use frame_support::traits::StorageVersion;
+    use frame_system::pallet_prelude::*;
+
+    #[pallet::config]
+    pub trait Config:
+        frame_system::Config + common::Config + permissions::Config + technical::Config
+    {
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        type WeightInfo: WeightInfo;
+
+        type AssetInfoProvider: AssetInfoProvider<
+            AssetIdOf<Self>,
+            AccountIdOf<Self>,
+            AssetSymbol,
+            AssetName,
+            BalancePrecision,
+            ContentSource,
+            Description,
+        >;
+    }
+
+    /// The current storage version.
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(<T as Config>::WeightInfo::regulate_asset())]
+        pub fn regulate_asset(origin: OriginFor<T>, asset_id: AssetIdOf<T>) -> DispatchResult {
+            // validate
+            let who = ensure_signed(origin)?;
+            <T as Config>::AssetInfoProvider::ensure_asset_exists(&asset_id)?;
+            ensure!(
+                <T as Config>::AssetInfoProvider::is_asset_owner(&asset_id, &who),
+                <Error<T>>::OnlyAssetOwnerCanRegulate
+            );
+            ensure!(
+                !Self::regulated_asset(asset_id),
+                <Error<T>>::AssetAlreadyRegulated
+            );
+
+            // act
+            <RegulatedAsset<T>>::set(asset_id, true);
+            Self::deposit_event(Event::AssetRegulated { asset_id });
+
+            Ok(())
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::issue_sbt())]
+        pub fn issue_sbt(
+            origin: OriginFor<T>,
+            symbol: AssetSymbol,
+            name: AssetName,
+            initial_supply: Balance,
+            allowed_assets: Vec<AssetIdOf<T>>,
+            description: Option<Description>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            // Check permission `who` can issue SBT
+            Permissions::<T>::check_permission(who.clone(), ISSUE_SBT)?;
+
+            let sbt_asset_id = T::AssetManager::register_from(
+                &who,
+                symbol,
+                name.clone(),
+                DEFAULT_BALANCE_PRECISION,
+                initial_supply,
+                true,
```

Comment:
SBT by its nature is not transferable. So yes it's only mintable and burnable by only the asset owner.

Reply:
Because SBT isn't transferable, The issuer won't be able to use the initial supply. 
So, I omitted the initial supply from extrinsic parameters and when issuing a new SBT,  Initial supply will be zero. 

Note: I re-ran benchmarking and added the new weights.

@wer1st @vovac12

Reply:
Ok